module 'pysam.libcalignedsegment' has no attribute 'CMATCH'

I'm getting a strange pysam error that only seems to occur when I am running with Python 3.8 on on MacOS latest in this [github action](https://github.com/tanghaibao/jcvi/blob/main/.github/workflows/pytest.yml). Error does not occur on ubuntu.

When importing pysam I get this error:

```
/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/site-packages/pysam/__init__.py:21: in <module>
    from pysam.libcalignedsegment import *
E   AttributeError: module 'pysam.libcalignedsegment' has no attribute 'CMATCH'
```

Versions:

- Python version: 3.8.10 (v3.8.10:3d8993a744, May  3 2021, 09:09:08) 
- [Clang 12.0.5 (clang-1205.0.22.9)]
- Pysam: pysam-0.23.0-cp38-cp38-macosx_11_0_universal2.whl 

Here is the log of the github action where this is failing: https://github.com/tanghaibao/jcvi/actions/runs/14951392044/job/42001281341

I don't see how [libcalignedsegment](https://github.com/pysam-developers/pysam/blob/master/pysam/libcalignedsegment.pyx) could be missing that attribute.

Maybe similar to https://github.com/pysam-developers/pysam/issues/1055 ?

--- a/pysam/libcalignedsegment.pyx
+++ b/pysam/libcalignedsegment.pyx
@@ -3409,10 +3409,16 @@ cpdef enum SAM_FLAGS:
     FSUPPLEMENTARY = 2048
 
 
+# TODO Remove these and remove the enumerators from __all__
+globals().update(getattr(CIGAR_OPS, "__members__"))
+globals().update(getattr(SAM_FLAGS, "__members__"))
+
+
 __all__ = [
     "AlignedSegment",
     "PileupColumn",
     "PileupRead",
+    "CIGAR_OPS",
     "CMATCH",
     "CINS",
     "CDEL",
@@ -3423,6 +3429,7 @@ __all__ = [
     "CEQUAL",
     "CDIFF",
     "CBACK",
+    "SAM_FLAGS",
     "FPAIRED",
     "FPROPER_PAIR",
     "FUNMAP",

--- a/setup.py
+++ b/setup.py
@@ -160,7 +160,7 @@ def write_configvars_header(filename, ext, prefix):
     log.info("creating %s for '%s' extension", filename, ext.name)
     with open(filename, "w") as outf:
         for var, value in config.items():
-            outf.write('#define {}_{} "{}"\n'.format(prefix, var, value))
+            outf.write(f'#define {prefix}_{var} "{value}"\n')
 
 
 @contextmanager
@@ -170,12 +170,12 @@ def set_compiler_envvars():
         if var in os.environ:
             if var == 'CFLAGS' and 'CCSHARED' in sysconfig.get_config_vars():
                 os.environ[var] += ' ' + sysconfig.get_config_var('CCSHARED')
-            print("# pysam: (env) {}={}".format(var, os.environ[var]))
+            print(f"# pysam: (env) {var}={os.environ[var]}")
         elif var in sysconfig.get_config_vars():
             value = sysconfig.get_config_var(var)
             if var == 'CFLAGS' and 'CCSHARED' in sysconfig.get_config_vars():
                 value += ' ' + sysconfig.get_config_var('CCSHARED')
-            print("# pysam: (sysconfig) {}={}".format(var, value))
+            print(f"# pysam: (sysconfig) {var}={value}")
             os.environ[var] = value
             tmp_vars += [var]
 
@@ -200,8 +200,7 @@ def configure_library(library_dir, env_options=None, options=[]):
         env_options = "--disable-bz2"
 
     if not os.path.exists(configure_script):
-        raise ValueError(
-            "configure script {} does not exist".format(configure_script))
+        raise ValueError(f"configure script {configure_script!r} does not exist")
 
     with changedir(library_dir), set_compiler_envvars():
         if env_options is not None:
@@ -311,7 +310,7 @@ class cy_build_ext(build_ext):
             ext.extra_link_args += ['-dynamiclib',
                                     '-rpath', '@loader_path',
                                     '-Wl,-headerpad_max_install_names',
-                                    '-Wl,-install_name,%s' % library_path,
+                                    f'-Wl,-install_name,{library_path}',
                                     '-Wl,-x']
         else:
             if not ext.extra_link_args:
@@ -393,24 +392,21 @@ config_headers = ["samtools/config.h",
 # the .pyx files. If no cython is available, the C-files included in the
 # distribution will be used.
 if HAVE_CYTHON:
-    print("# pysam: cython is available - using cythonize if necessary")
+    print(f"# pysam: Cython {cython.__version__} is available - using cythonize if necessary")
     source_pattern = "pysam/libc%s.pyx"
 else:
-    print("# pysam: no cython available - using pre-compiled C")
+    print("# pysam: no Cython available - using pre-compiled C")
     source_pattern = "pysam/libc%s.c"
 
 # Exit if there are no pre-compiled files and no cython available
 fn = source_pattern % "htslib"
 if not os.path.exists(fn):
     raise ValueError(
-        "no cython installed, but can not find {}."
-        "Make sure that cython is installed when building "
-        "from the repository"
-        .format(fn))
+        f"no Cython installed, but cannot find {fn}. "
+        "Make sure that Cython is installed when building from the repository")
 
-print("# pysam: htslib mode is {}".format(HTSLIB_MODE))
-print("# pysam: HTSLIB_CONFIGURE_OPTIONS={}".format(
-    HTSLIB_CONFIGURE_OPTIONS))
+print(f"# pysam: htslib mode is {HTSLIB_MODE}")
+print(f"# pysam: HTSLIB_CONFIGURE_OPTIONS={HTSLIB_CONFIGURE_OPTIONS}")
 htslib_configure_options = None
 
 if HTSLIB_MODE in ['shared', 'separate']:
@@ -425,8 +421,7 @@ if HTSLIB_MODE in ['shared', 'separate']:
          "--disable-libcurl"])
 
     HTSLIB_SOURCE = "builtin"
-    print("# pysam: htslib configure options: {}".format(
-        str(htslib_configure_options)))
+    print(f"# pysam: htslib configure options: {htslib_configure_options}")
 
     config_headers += ["htslib/config.h"]
     if htslib_configure_options is None:
@@ -441,7 +436,7 @@ if HTSLIB_MODE in ['shared', 'separate']:
         htslib_make_options = run_make_print_config()
 
     for key, value in htslib_make_options.items():
-        print("# pysam: htslib_config {}={}".format(key, value))
+        print(f"# pysam: htslib_config {key}={value}")
 
     external_htslib_libraries = ['z']
     if "LIBS" in htslib_make_options:
@@ -478,11 +473,11 @@ elif HTSLIB_MODE == 'shared':
     htslib_library_dirs = ["."] # when using setup.py develop?
     htslib_include_dirs = ['htslib']
 else:
-    raise ValueError("unknown HTSLIB value '%s'" % HTSLIB_MODE)
+    raise ValueError(f"unknown HTSLIB value {HTSLIB_MODE!r}")
 
 # build config.py
 with open(os.path.join("pysam", "config.py"), "w") as outf:
-    outf.write('HTSLIB = "{}"\n'.format(HTSLIB_SOURCE))
+    outf.write(f'HTSLIB = "{HTSLIB_SOURCE}"\n')
     config_values = collections.defaultdict(int)
 
     if HTSLIB_SOURCE == "builtin":
@@ -502,8 +497,8 @@ with open(os.path.join("pysam", "config.py"), "w") as outf:
                         "HAVE_LIBDEFLATE",
                         "HAVE_LIBLZMA",
                         "HAVE_MMAP"]:
-                outf.write("{} = {}\n".format(key, config_values[key]))
-                print("# pysam: config_option: {}={}".format(key, config_values[key]))
+                outf.write(f"{key} = {config_values[key]}\n")
+                print(f"# pysam: config_option: {key}={config_values[key]}")
 
 # create empty config.h files if they have not been created automatically
 # or created by the user:
@@ -542,13 +537,15 @@ if os.environ.get("CIBUILDWHEEL", "0") == "1":
 suffix = sysconfig.get_config_var('EXT_SUFFIX')
 
 internal_htslib_libraries = [
-    os.path.splitext("chtslib{}".format(suffix))[0]]
+    os.path.splitext(f"chtslib{suffix}")[0],
+    ]
 internal_samtools_libraries = [
-    os.path.splitext("csamtools{}".format(suffix))[0],
-    os.path.splitext("cbcftools{}".format(suffix))[0],
+    os.path.splitext(f"csamtools{suffix}")[0],
+    os.path.splitext(f"cbcftools{suffix}")[0],
     ]
 internal_pysamutil_libraries = [
-    os.path.splitext("cutils{}".format(suffix))[0]]
+    os.path.splitext(f"cutils{suffix}")[0],
+    ]
 
 libraries_for_pysam_module = external_htslib_libraries + internal_htslib_libraries + internal_pysamutil_libraries
 

